deploy/eck-elasticsearch/examples/hot-warm-cold.yaml

```diff
@@ -0,0 +1,134 @@
+---
+nodeSets:
+- name: hot
+  count: 1
+  config:
+    node.roles: ["data_hot", "data_content", "ingest", "master"]
+    node.store.allow_mmap: false
+  podTemplate:
+    spec:
+      containers:
+      - name: elasticsearch
+        resources:
+          limits:
+            memory: 16Gi
+            cpu: 4
+      # Affinity/Anti-affinity settings for controlling the 'spreading' of Elasticsearch
+      # pods across existing hosts.
+      # ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+      # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-advanced-node-scheduling.html#k8s-affinity-options
+      #
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: beta.kubernetes.io/instance-type
+                operator: In
+                # This should be adjusted to the instance type according to your setup
```

Comment:
One issue I see with comments like this is that they are not present in the rendered result. So a user blindly applying the values file an expecting a working configuratin to result from that would be somewhat surprised.